lib/bankaccounts/savings_account.rb

```diff
@@ -0,0 +1,39 @@
+module BankAccounts
+
+  class SavingsAccount < Account # inherits from Account class
+
+    def initialize(id, initial_balance)
+      super
+      # raises an error if initial_balance is less than 10
+      if initial_balance < 10
+        raise ArgumentError.new("Initial balances must be at least 10")
+      end
```

Comment:
Good logic :+1: Suggestion: Have you thought about extracting this into a separate method that can be called from `initialize`?